lib/rspec/mocks/example_methods.rb

```diff
@@ -201,6 +201,14 @@ def allow_message_expectations_on_nil
         RSpec::Mocks.space.proxy_for(nil).warn_about_expectations = false
       end
 
+      # Raise an error when setting expectations on nil.
+      #
+      # By default warning messages are issued when expectations are set on
+      # nil.  This is to force test failures when an expectation is set on nil.
+      def disallow_message_expectations_on_nil
```

Comment:
@myronmarston That is exactly where I was expecting to find this when I was rooting around looking to mimic the existing code for `warn_about_expectations`, myself. I would like it as a global config, as well as with `warn`. Seems to make more sense as a config option than to have to set it for each example.